rules.js

```diff
@@ -186,6 +186,22 @@ class sub_with_return extends rule(3) {
     }
 }
 
+class yoda_condition extends rule(3) {
+    constructor() {
+        super()
+        this._node = 'bop'
+        this.message = `in conditional expression constant should be on the right`
+    }
+
+    check(node) {
+        const compareOp = /^[>=<]{1,2}$/.test(node.op)
+        const leftConst = /const|number|string/.test(node.left.node)
```

Comment:
i don't exactly get this test. is it a regular checking for the node's type?

Reply:
Just a check if left node is constant (true,false,invalid), string or number. Better than `node.left.node === 'const' || node.left.node === 'string'...`

Reply:
okay interesting so `.node` is a string rep of the type?

Reply:
Yes